app/src/main/java/com/example/habitsmasher/HabitEventList.java

```diff
@@ -0,0 +1,43 @@
+package com.example.habitsmasher;
+
+import android.net.Uri;
+
+import java.util.ArrayList;
+import java.util.Date;
+
+/**
+ * This is the Habit Event List Class.
+ * It's purpose is to store habit events
+ */
+public class HabitEventList extends ArrayList{
+    private ArrayList<HabitEvent> _habitEvents = new ArrayList<>();
+
+    /**
+     * Gets the list of habit events
+     * @return habitEvents: The list of habit events
+     */
+    public ArrayList<HabitEvent> getHabitEvents() {
+        return this._habitEvents;
+    }
+
+    /**
+     * Creates a new habit event and add its to the habit event list
+     * @param startDate (Date): The start date of the habit event to add
+     * @param comment (String): The comment of the habit event to add
+     * @param pictureUri (String): The URL of the picture of the habit event to add
+     */
+    public void addHabitEvent(Date startDate, String comment, Uri pictureUri) {
+        // Create habit event and add it to the list
+        HabitEvent eventToAdd = new HabitEvent(startDate, comment, pictureUri);
+        _habitEvents.add(eventToAdd);
+    }
+
+    /**
+     * Adds a habit event to the habit event list
+     * @param eventToAdd (HabitEvent): The event to add to the habit event list
+     */
+    public void addHabitEvent(HabitEvent eventToAdd) {
```

Comment:
This method had the same name as the comment I left above, if both are meant to stay, I think renaming them would be a good idea so they are not as confusing

Reply:
I think they are both useful, they are the same name, but **different parameters**. This gives us the flexibility to either provide the `HabitEvent` object to the add method or supply each field separately. As long as we test both, they can both stay in my opinion.

Reply:
I would agree with Rudy. It's the same function however with different parameters, so I don't think a different name would make a lot of sense